lib/install/validate/validator.go

```diff
@@ -361,8 +363,36 @@ func (v *Validator) compatibility(ctx context.Context, conf *config.VirtualConta
 		return
 	}
 
+	// check session's datastore(s) exist
 	_, err := v.Session.Datastore.AttachedClusterHosts(v.Context, v.Session.Cluster)
 	v.NoteIssue(err)
+
+	// closure to look up a datastore based on a URL from the config & check to see if it's writeable
+	checkDS := func(u *url.URL) {
+		datastores, err := v.Session.Finder.DatastoreList(ctx, fmt.Sprintf("[%s] %s", u.Host, u.Path))
+		if err != nil {
+			v.NoteIssue(err)
+		}
+		if len(datastores) != 1 {
+			// this is basically an assert; at this point DatastoreList should absolutely only return one result
+			v.NoteIssue(errors.Errorf("Looking up datastore %s returned %d results.", u.String(), len(datastores)))
+		}
+		if !v.isTargetDatastoreWriteable(ctx, datastores[0]) {
+			v.NoteIssue(errors.Errorf("Datastore %s is not writeable by the compute resource provided by --compute-resource", u.String()))
```

Comment:
Basically, I think the logic is that datastore not connected to host, so the datastore is not writable. But I assume "not writable" might not a VI admin term.